src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java

```diff
@@ -423,6 +443,94 @@ private static String topLevelType(String type) {
         return topLevelType;
     }
 
+    /**
+     * Retrieves all valid reference tags based on their Javadoc type.
+     *
+     * @param tags        the tag to validate
+     * @param javadocTagType type of tag to check
+     * @return a {@code List} of valid Javadoc tags based on the specified type
+     */
+    public static List<JavadocTag> getValidReferenceTags(
+            Collection<JavadocTag> tags, JavadocUtil.JavadocTagType javadocTagType) {
+
+        return tags.stream()
+            .filter(tag -> javadocTagType != JavadocUtil.JavadocTagType.BLOCK || !tag.isInlineTag())
```

Comment:
this condition is supper confusing.

we hardcoding value of javadocTagType 
and filtering by it, some nonsense is there.
```
       final List<JavadocTag> blockTags = getValidTags(textBlock,
                JavadocUtil.JavadocTagType.BLOCK);
        validTags.addAll(getValidReferenceTags(blockTags,
                JavadocUtil.JavadocTagType.BLOCK));
```

you should move it filtering to this method:
```
private static List<JavadocTag> getValidTags(TextBlock cmt,
            JavadocUtil.JavadocTagType javadocTagType) {
        return JavadocUtil.getJavadocTags(cmt, javadocTagType).getValidTags();
    }
```

so list of tags will be valid and we do not need not trust it.